File: shared/src/main/java/ru/privatenull/pnauth/api/AuthSessionApi.java
package ru.privatenull.pnauth.api;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

/** Authentication lifecycle operations exposed to platform adapters. */
public interface AuthSessionApi {
    CompletableFuture<AuthStatus> onJoin(UUID uniqueId, String username);

    default CompletableFuture<AuthStatus> onJoin(UUID uniqueId, String username, String ip) {
        return onJoin(uniqueId, username);
    }

    void onQuit(UUID uniqueId);

    CompletableFuture<AuthResult> register(UUID uniqueId, String username, String password, String confirmation);

    CompletableFuture<AuthResult> login(UUID uniqueId, String password);

    CompletableFuture<AuthResult> logout(UUID uniqueId);

    CompletableFuture<AuthResult> changePassword(UUID uniqueId, String oldPassword, String newPassword);

    AuthStatus status(UUID uniqueId);

    boolean isAuthenticated(UUID uniqueId);
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/api/AuthSessionApi.kt
package ru.privatenull.pnauth.api

import java.util.UUID
import java.util.concurrent.CompletableFuture

/** Authentication lifecycle operations exposed to platform adapters. */
interface AuthSessionApi {
    fun onJoin(uniqueId: UUID, username: String): CompletableFuture<AuthStatus>

    fun onJoin(uniqueId: UUID, username: String, ip: String?): CompletableFuture<AuthStatus> {
        return onJoin(uniqueId, username)
    }

    fun onQuit(uniqueId: UUID)

    fun register(uniqueId: UUID, username: String, password: String, confirmation: String): CompletableFuture<AuthResult>

    fun login(uniqueId: UUID, password: String): CompletableFuture<AuthResult>

    fun logout(uniqueId: UUID): CompletableFuture<AuthResult>

    fun changePassword(uniqueId: UUID, oldPassword: String, newPassword: String): CompletableFuture<AuthResult>

    fun status(uniqueId: UUID): AuthStatus

    fun isAuthenticated(uniqueId: UUID): Boolean
}
